### ISSUE TITLE
`appsody init <stack> none` is not intuitive

### ISSUE THREAD

**Issue by ianpartridge** [appsody/appsody#374]
Recently I demoed Appsody at the try! Swift NYC conference, and also to an internal team of developers.

In both cases I was "appsody-ifying" an existing application by using `appsody init <stack> none`.

I consistently got feedback that it was not intuitive/obvious what was going on here - the special magic of "none" is confusing.

I'm opening this to discuss whether an improvement might be made.

As a straw man proposal, perhaps we could have a new `appsody enable <stack>` command for appsody-ifying an existing application.

Or, perhaps `appsody init <stack>` could behave the same as `appsody init <stack> none` if the current directory is not empty.

**Comment by pushkarnk** [comment]
@gireeshpunathil and I also discussed this today. To "appsodify" a project that corresponds to one of the existing appsody stacks, it must be enough to issue an `appsody init` (with no other arguments). The CLI should be able to study the file structure of the project and decide which stack is to be used.

**Comment by pushkarnk** [comment]
> @gireeshpunathil and I also discussed this today. To "appsodify" a project that corresponds to one of the existing appsody stacks, it must be enough to issue an `appsody init` (with no other arguments). The CLI should be able to study the file structure of the project and decide which stack is to be used.

I've heard a similar capability exists in [source-to-image/s2i](https://github.com/openshift/source-to-image)?

**Comment by ebullient** [comment]
`ibmcloud dev enable` is what the public team does to enable an existing project. We might learn from that. 

`appsody init <stack> enable` or `appsody init <stack> convert` might be better options.

**Comment by kylegc** [comment]
Yes "inferring" the stack would be very difficult so I don't think we should go down that path. However even if the user specifies the stack, it may still be difficult to "transform" the existing project into a form the the given stack accepts. We would like to do at least some of this "transformation" for the user, but are still working out how that will be done. (Related: #372 ) 

For now, we just provide a way to setup the `.appsody-config.yaml` on an existing project and leave the transformation up to the user.

If possible, I'd like to stick to a single `init` command for both new and transforming, but I agree the `none` argument is confusing. Honestly I liked the `--no-template` flag better. Here are a couple more options:

1) Make the template parameter always a flag instead of a positional argument. Such as `appsody init <stack> --template=<template>`. Then `--template=none` would be clearer, and we could still use the default if it wasn't specified.

2) Change the way init acts when existing project files are detected. Such that running init on an empty directory would apply the template, but running init on a dir with existing project files would not apply the template.

**Comment by nastacio** [comment]
I always assumed that using `init` was how one converted an existing app to an appsody app.

> "Or, perhaps appsody init <stack> could behave the same as appsody init <stack> none if the current directory is not empty."

Requiring the extra parameter for init <stack> feels like a welcome safety measure. I think the error message is missing the explanation about the alternatives after the failure.

```
[Error] Non-empty directory found with files which may conflict with the template project.
It is recommended that you run `appsody init <stack>` in an empty directory.
If you wish to proceed and possibly overwrite files in the current directory, try again with the --overwrite option.
[Error] non-empty directory found with files which may conflict with the template project

```
Instead of the line “It is recommended …“, which is not quite a recommendation, but rather an oversimplification of the alternatives, we could have:

“If the current directory contains a project you wish to convert to an appsody project, use the stack name “none” to create the required appsody files.”

PS: Also note the duplicate echoing of the same [Error] message in the current CLI.

**Comment by nastacio** [comment]
> `ibmcloud dev enable` is what the public team does to enable an existing project. We might learn from that.
> 
> Some options based on that:
> `appsody init <stack> enable`
> `appsody init <stack> convert`
> or maybe something more direct like:
> `appsody enable <stack>`
> `appsody convert <stack>`.
> 

I like the suggestion of a new "enable" command.

As a smaller observation, and for consistency with the current CLI syntax, if we are sticking with the `init` command, this new conversion instruction should be a flag, rather than an overload the template parameter:

Usage:
  appsody init [stack] or [repository]/[stack] **[template]** [flags]

PS: It follows that specifying a template and a would-be "--enable" flag should be an error condition.

**Comment by knolleary** [comment]
One thing that threw me with the log output of both `--no-template` and `none` was it saying it was downloading and extracting files from a template tgz even though I has asked it not to use a template.

```
$ appsody init local/node-red none
Running appsody init...
Downloading node-red template project from file:///Users/nol/code/knolleary/appsody-node-red-stack/repo/node-red.templates.simple.tar.gz
Download complete. Extracting files from node-red.tar.gz
Setting up the development environment
```

On the more substantive issue, I found `--no-template` a clearer and more intuitive way indicate my intent, rather than have `none` as a positional parameter.

I also think it is clearer having a single entry point to get started with appsody, whether starting from scratch or with existing code - in other words, keeping it all under `appsody init`.

**Comment by knolleary** [comment]
@kylegc #428 raised
